WooCommerce/Classes/ViewRelated/Products/Filters/FilterProductListViewModel.swift

```diff
@@ -136,15 +139,53 @@ extension FilterProductListViewModel.ProductListFilter {
             return FilterTypeViewModel(title: title,
                                        listSelectorConfig: .staticOptions(options: options),
                                        selectedValue: filters.productStatus)
-        case .productType:
-            let options: [ProductType?] = [nil, .simple, .variable, .grouped, .affiliate, .subscription, .variableSubscription, .bundle, .composite]
+        case let .productType(siteID):
+            let options = buildPromotableTypes(siteID: siteID, storageManager: storageManager)
             return FilterTypeViewModel(title: title,
                                        listSelectorConfig: .staticOptions(options: options),
-                                       selectedValue: filters.productType)
+                                       selectedValue: filters.promotableProductType)
         case let .productCategory(siteID):
             return FilterTypeViewModel(title: title,
                                        listSelectorConfig: .productCategories(siteID: siteID),
                                        selectedValue: filters.productCategory)
         }
     }
+
+    /// Builds the products types filter array identifying which extension is available or not.
+    ///
+    private func buildPromotableTypes(siteID: Int64, storageManager: StorageManagerType) -> [PromotableProductType?] {
+        let activePluginNames = fetchActivePluginNames(siteID: siteID, storageManager: storageManager)
+        let isSubscriptionsAvailable = Set(activePluginNames).intersection(SitePlugin.SupportedPlugin.WCSubscriptions).count > 0
+        let isCompositeProductsAvailable = activePluginNames.contains(SitePlugin.SupportedPlugin.WCCompositeProducts)
+        let isProductBundlesAvailable = activePluginNames.contains(SitePlugin.SupportedPlugin.WCProductBundles)
```

Comment:
Non-blocking: how about checking against the plugin path instead of the name, which is more subject to changes? There's a way to find a WC plugin path linked at p7Ldg5-1Xd-p2

Reply:
Good point. I think it would be better to move all plugin name matches to its path in one go. Maintaining both approaches can be confusing and it gets difficult to enforce across multiple teams.

Reply:
I added a task for it https://github.com/woocommerce/woocommerce-ios/issues/10775